File: wlr/OGUR/OGUR/Strategies/StraightLineStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OGUR.Creatures;
using SPX.Entities;
using SPX.Core;

namespace OGUR.Strategies
{
    class StraightLineStrategy:IStrategy
    {
        public StraightLineStrategy(ICreature parent, params int[] targetTypes)
            : base(parent)
        {
            foreach (var targetType in targetTypes)
            {
                _targets.AddTargetTypes(targetType);
            }
        }

        public override void Act()
        {
            Console.WriteLine(_parent.Get(StatType.MOVE_COOL_DOWN));
            _parent.MoveIfPossible(0, 1);
        }
    }
}

File: proj/src/aigilas/strategies/impl/StraightLineStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.ICreature;
import aigilas.strategies.IStrategy;
import aigilas.strategies.Strategy;
import spx.bridge.ActorType;
import spx.core.Settings;

public class StraightLineStrategy extends IStrategy {
    public StraightLineStrategy(ICreature parent, ActorType... targetTypes)

    {
        super(parent, Strategy.StraightLine);

        for (ActorType targetType : targetTypes) {
            _targets.addTargetTypes(targetType);
        }
    }

    @Override
    public void act() {
        _parent.moveIfPossible(0, 1);
        if (_parent.getLocation().GridY >= Settings.get().tileMapHeight - 1) {
            _parent.setInactive();
        }
    }
}
